hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+                sb_position_temp = sb_position.data
+        else:
+            sb_position_temp = sb_position.deepcopy()
+        #
+
+        # Parsing sideband size
+        if sb_size is None:  # Default value is 1/2 distance between sideband and central band
+            if reference is None:
+                sb_size = self.find_sideband_size(sb_position)
+            else:
+                sb_size = reference.find_sideband_size(sb_position)
+        else:
+            if not isinstance(sb_size, BaseSignal):
+                if isinstance(sb_size, np.ndarray) and sb_size.size > 1:  # transpose if np.array of multiple instances
+                    sb_size = BaseSignal(sb_size).T
+                else:
+                    sb_size = BaseSignal(sb_size)
+
+        if sb_size.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_size.axes_manager.navigation_size:
+                warnings.warn('Sideband size dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_size_temp = np.float64(sb_size.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_size_temp = np.float64(sb_size.data)
+        else:
+            sb_size_temp = sb_size.deepcopy()
+        #
+
+        # Standard edge smoothness of sideband aperture 5% of sb_size
+        if sb_smoothness is None:
+            sb_smoothness = sb_size * 0.05
+        else:
+            if not isinstance(sb_smoothness, BaseSignal):
+                if isinstance(sb_smoothness, np.ndarray) and sb_smoothness.size > 1:
+                    sb_smoothness = BaseSignal(sb_smoothness).T
+                else:
+                    sb_smoothness = BaseSignal(sb_smoothness)
+
+        if sb_smoothness.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_smoothness.axes_manager.navigation_size:
+                warnings.warn('Sideband smoothness dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_smoothness_temp = np.float64(sb_smoothness.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_smoothness_temp = np.float64(sb_smoothness.data)
+        else:
+            sb_smoothness_temp = sb_smoothness.deepcopy()
+
+        # Convert sideband size from 1/nm or mrad to pixels
+        if sb_unit == 'nm':
+            sb_size_temp = sb_size_temp / np.mean(self.f_sampling)
+            sb_smoothness = sb_smoothness / np.mean(self.f_sampling)
+        elif sb_unit == 'mrad':
+            try:
+                ht = self.metadata.Acquisition_instrument.TEM.beam_energy
+            except AttributeError:
+                ht = int(input('Enter beam energy in kV: '))
```

Comment:
That's a good point. We should probably have a `TEM` superclass from which all `TEM` signals inherit that includes `set_microscope_parameters`. It should work for EELS though. If you get a `NotImplementedError` is probably because you didn't initialize the backend before running the function e.g. `%matplotlib qt4`.

I don't think that it is a good idea to ask for input interactively. For example, [HyperSpyUI](http://hyperspy.org/hyperspyUI/) would struggle with it. If it raises an error, the user can then use `set_microscope_parameters` that includes a UI for HyperSpyUI. This is what we currently do for other methods that require missing, [here is an example](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/_signals/eels.py#L1050).